data/utilities.lua

```diff
@@ -0,0 +1,119 @@
+
+---@class PrototypeBase https://wiki.factorio.com/PrototypeBase
+---@class IconSpecification https://wiki.factorio.com/Types/IconSpecification
+

```

Comment:
Please follow the [style guide's bit on modules](https://github.com/luarocks/lua-style-guide#modules). In short, create a local table that you then assign all the functions to, then return that table:

```LUA
local utilities = {}

function utilities.copy_prototype(...)

...

return utilities

Reply:
removed data.lua